source/lac/scalapack.cc

```diff
@@ -27,7 +27,7 @@ DEAL_II_NAMESPACE_OPEN
 template <typename NumberType>
 ScaLAPACKMatrix<NumberType>::ScaLAPACKMatrix(const size_type n_rows_,
                                              const size_type n_columns_,
-                                             const std::shared_ptr<const Utilities::MPI::ProcessGrid> process_grid,
+                                             const std::shared_ptr<const Utilities::MPI::ProcessGrid> &process_grid,
```

Comment:
@davydden explained to me in a similar situation why it makes sense to pass this by value, but I don't remember the details. @davydden should we pass by value here?

Reply:
I don’t remember the argument why, but passing shared pointer by values is certainly ok

Reply:
Admittedly, the performance impact is probably negligible indeed. 

On the other hand, we are just doing extra work without any reason. We are first copying when the function is called and then again when we we copy the `shared_ptr` to a member variable. 
Using a reference instead avoids the first copy.
A different approach would be to call by value (not `const`) and `std::move` into the member variable.

Copying (without moving) just seems to be conceptionally inappropriate here.

Reply:
Since it’s being copied into a member variable anyway, I agree with you

Reply:
I think there is no reason not to make this a reference. In the best of cases, `shared_ptr` is a small class so that copying it is cheap -- but copying also implies incrementing the use counter of the object pointed to, and when the copy goes out of scope, counting it down by one again. This may imply using a mutex.